### DISCUSSION TITLE
clang messages in vs code, but I use gcc

### DISCUSSION THREAD

**Discussion by Vdaleke** [clangd/clangd#2222] [Q&A] [answered]
I'm building project under g++-14 on macOS (Apple Silicon). 

Project using CMake, so, I generated compile_command.json, where each command use g++-14.
[compile_commands.json](https://github.com/user-attachments/files/17802248/compile_commands.json)

But but i keep getting messages that clang can't find g++ extensions in my project like this: 
![image](https://github.com/user-attachments/assets/a6555557-06d2-469f-a58c-2080e6f4d37a)

I tried to specify `.clangd` like this: 
```
CompileFlags:
  Compiler: g++-14
```
But it didn't help. Why does this problem occur and how to fix it? Thank you, hope for your answer.

clangd logs:
[5-clangd.log](https://github.com/user-attachments/files/17802764/5-clangd.log)

**Comment by HighCommander4** [comment, marked as answer]
Clangd is a clang-based tool; it always uses the clang frontend to parse your code. The `Compiler` setting is only used to help locate the correct system headers, which can vary between compilers.

So, for clangd to work well, the code needs to be able to compile with clang.

I know clang supports many gcc extensions; I'm not sure whether `__float128` is one of them. Folks in the [clang user forum](https://discourse.llvm.org/c/clang/using-clang/47) might have more specific advice for you about that.
